lightning/src/ln/channelmanager.rs

```diff
@@ -346,6 +352,14 @@ pub struct ChannelManager<ChanSigner: ChannelKeys> {
 	channel_state: Mutex<ChannelHolder<ChanSigner>>,
 	our_network_key: SecretKey,
 
+	/// Per-peer state storage.
+	/// Because adding or removing an entry is rare, we usually take an outer read lock and then
+	/// operate on the inner value freely. Sadly, this prevents parallel operation when opening a
+	/// new channel.
+	/// If we are connected to a peer we always at least have an entry here, to store their features
+	/// so that we have them available if we open a channel with them and need it for routing.
```

Comment:
This is a bit verbose and may get out of sync with `PerPeerState` by talking about specific data within it. I'd recommend rewording more succinctly as:

> Per-peer state used for opening channels and routing.

This can also replace the first sentence of the documentation since it mostly just restates the field name.

Reply:
Hmm, I mean where features gets stored, sure, I suppose, though the note that "we'll always have something here if we're connected" is an important thing to document - it indicates when you can rely on an entry vs having to check and see if there is one. I'll just drop the "to store their features..." bit.

Reply:
My larger point is that if it is important then it should be stated upfront in the documentation (rather than after the locking comments) and done in a succinct manner such that the reader can quickly understand its purpose. IMHO, "Per-peer state storage" doesn't accomplish that as it's essentially restating the field name.

Reply:
Right, ok, missed the objection to the opening line. I reworded it to "The bulk of our storage will eventually be here (channels and message queues and the like)." I presume thats a bit more descriptive?

Reply:
Eh, I'd expect the first sentence to use the word "peer" at very least; "storage" doesn't seem like a primary characteristic of what the field contains. And why is "bulk" relevant to the reader? The fact that this data is persisted seems like an ancillary detail to me. More important is how it is used.

Reply:
I mean the variable name is pretty clear about it being a peer-indexed object :). As for "storage" its more of a future-looking thing - once we move channels and pending_msg_events, its really almost all of the ChannelManager "stuff" lives there.
I'm not sure what you're referring to about "persisted" there - nothing mentions persistence, only that a precondition exists that there is an entry if the peer is connected (ie if you're in a message handler, its guaranteed to be there, so you don't have to check).

Reply:
Generally speaking, documentation should primarily state what something is and how it is used (i.e., its purpose). This is the first thing that should be stated. In other words, we are storing it only because it has some primary purpose. By "persisted" I simply meant saved to storage, which seems like an ancillary detail to the fact that this has some purpose.

I can see the argument that some of this documentation belongs on `PeerState` while other belongs on `per_peer_state`. I would like to see a clear divide between the two. :) In that sense, yes, I see your point: this should mostly talk about locking and how the mapping works. But I'd still argue that storage is not the primary purpose and therefore should not be in the first sentence.

Separately, anything future-looking should not be part of documentation. Just put it in once the code reflects that reality. Or at very least it should be more of an aside rather than the primary sentence in my opinion.